src/sentry/tasks/commit_context.py

```diff
@@ -46,7 +49,12 @@ def queue_comment_task_if_needed(commit: Commit, group_owner: GroupOwner):
         not pr.pullrequestcomment_set.exists()
         or group_owner.group_id not in pr.pullrequestcomment_set.get().group_ids
     ):
-        # TODO: Debouncing Logic
+        cache_key = DEBOUNCE_PR_COMMENT_CACHE_KEY(pullrequest_id=pr.id)
+        if cache.get(cache_key) is not None:
+            return
+
+        cache.set(cache_key, True, PR_COMMENT_TASK_TTL)
```

Comment:
Be careful about this - there's a race condition here where two processes fetch the key, find it's missing and then set the cache value and then both fire the task.

You can avoid this by checking whether the set succeeds, and then only fire the task if it does.

Reply:
would it be better to acquire a lock?

Reply:
Yeah, you can do that too. The locks work basically the same way